bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -55,7 +60,46 @@ public static String getLocalIpv4HostAddress() {
             }
             return hostAddress;
         } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: " + ex.getMessage(), ex);
+            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
+            return null;
+        }
+    }
+
+    /**
+     * Get all broadcast addresses on the current host
+     *
+     * @return list of broadcast addresses, empty list if no broadcast addresses found
+     */
+    public static List<String> getAllBroadcastAddresses() {
+        List<String> broadcastAddresses = new LinkedList<String>();
```

Comment:
Hm, shouldn't the formatter replace `new LinkedList<String>()` be replaced by an `Diamond Operator`?